core/store/src/trie/mod.rs

```diff
@@ -586,6 +609,23 @@ impl Trie {
         memory_usage
     }
 
+    fn delete_value(
+        &self,
+        memory: &mut NodesStorage,
+        value: &ValueHandle,
+    ) -> Result<(), StorageError> {
+        match value {
+            ValueHandle::HashAndSize(_, hash) => {
+                let bytes = self.storage.retrieve_raw_bytes(hash)?;
+                memory.refcount_changes.entry(*hash).or_insert_with(|| (bytes.to_vec(), 0)).1 -= 1;
```

Comment:
Is it guaranteed that this cannot underflow?

Reply:
it's signed i32 + we shouldn't have 2^31 storage operations in a block

Reply:
also the refcounts are stored only locally and are not part of the state